trieste/space.py

```diff
@@ -235,14 +223,14 @@ def __contains__(self, value: TensorType) -> bool | TensorType:
         :param value: A point to check for membership of this :class:`SearchSpace`.
         :return: `True` if ``value`` is a member of this search space, else `False`. May return a
             scalar boolean `TensorType` instead of the `bool` itself.
-        :raise ValueError (or InvalidArgumentError): If ``value`` has a different dimensionality
-            from the search space.
+        :raise ValueError (or tf.errors.InvalidArgumentError): If ``value`` has a different
+            dimensionality from the search space.
         """
-        if not shapes_equal(value, self._lower):
-            raise ValueError(
-                f"value must have same dimensionality as search space: {self._lower.shape},"
-                f" got shape {value.shape}"
-            )
+        tf.debugging.assert_equal(
+            shapes_equal(value, self._lower),
+            True,
+            message="value must have same dimensionality as search space"
+        )
```

Comment:
this is a bit of a hack, to get around the fact that `tf.debugging.assert_shapes` considers 0 and [0] to have the same shape